aptos-move/block-executor/src/view.rs

```diff
@@ -979,15 +979,7 @@ impl<'a, T: Transaction, S: TStateView<Key = T::Key>, X: Executable>
     }
 
     fn generate_delayed_field_id(&self) -> T::Identifier {
-        match &self.latest_view.latest_view {
-            ViewState::Sync(state) => (state.counter.fetch_add(1, Ordering::SeqCst) as u64).into(),
-            ViewState::Unsync(state) => {
-                let mut counter = state.counter.borrow_mut();
-                let id = (*counter as u64).into();
-                *counter += 1;
-                id
-            },
-        }
+        self.latest_view.generate_delayed_field_id()
```

Comment:
T::Identifier is a DelayedFieldID. 
`generate_delayed_field_id()` generates u32. Does the type casting work?

Reply:
on line 944 above is the function:

    fn generate_delayed_field_id(&self) -> Self::IdentifierV2 {

so no casting needed

Reply:
I am calling latest_view, not latest_view.latest_view